ciao-controller/internal/datastore/sqlite3db.go

```diff
@@ -1139,6 +1155,46 @@ func (ds *sqliteDB) updateWorkload(w types.Workload) error {
 	return nil
 }
 
+func (ds *sqliteDB) deleteWorkload(ID string) error {
+	db := ds.getTableDB("workload_template")
+
+	ds.dbLock.Lock()
+	defer ds.dbLock.Unlock()
+
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+
+	err = ds.deleteWorkloadDefault(tx, ID)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	err = ds.deleteWorkloadStorage(tx, ID)
+	if err != nil {
+		tx.Rollback()
+		return err
+	}
+
+	filename := fmt.Sprintf("%s_config.yaml", ID)
+	path := fmt.Sprintf("%s/%s", ds.workloadsPath, filename)
```

Comment:
As it currently stands this protects the workload in the database if the remove fails. I think we should follow @markdryan's suggestion. As if you're failing to delete it's probably because it's missing or some other problem so it will be better to delete it from the database than have an unlauchable workload as the required yaml file is missing.